src/Kiota.Builder/Extensions/OpenApiUrlTreeNodeExtensions.cs

```diff
@@ -26,6 +27,7 @@ public static class OpenApiUrlTreeNodeExtensions
                                                     .Select(CleanupParametersFromPath)
                                                     .Select(static (y, idx) => idx == 0 ? y : y.ToFirstCharacterUpperCase())))
                             ?.Select(static x => x.CleanupSymbolName())
+                            ?.Select(static x => GenerationConfiguration.ModelsNamespaceSegmentName.Equals(x, StringComparison.OrdinalIgnoreCase) ? $"{x}Requests" : x) //avoids projecting requests builders to models namespace
```

Comment:
Would it be possible to emit an error if `models` and `modelsRequests` are included in the description?

Reply:
That's the beauty of this solution. I don't think we need to. Everything that maps to the models path segment will start with models, everything that maps to the model requests path segment will start with models requests. Yes they'll be in the same namespace, but they shouldn't collide.